Automattic/wp-calypso
# Fix `Need more storage?` link on Hosting Overview page

Fixes https://github.com/Automattic/wp-calypso/issues/94670
Related to https://github.com/Automattic/wp-calypso/pull/94144

## Proposed Changes

On the Hosting Overview page, the plan storage widget is sometimes a div and sometimes an anchor. The `Need more storage?` text in the footer of that widget was previously always a link, which is wrong since we don't want nested links. https://github.com/Automattic/wp-calypso/pull/94144 sought to fix this but made it so `Need more storage?` is never a link instead. This PR fixes that.

I also took the opportunity to convert `client/blocks/plan-storage/index.js` to Typescript.

## Why are these changes being made?


Because we don't want nested anchors (since that's not valid HTML).

## Testing Instructions



1. Have an Atomic site without storage addons (that's not enrolled in A4A)
2. Navigate to `/overview/:site_slug`
3. Ensure that `Need more storage?` is a clickable link
4. Have a free site without storage addons
5. Navigate to `/overview/:site_slug`
6. Ensure that `Need more storage?` is not a link
